### PR TITLE
Fix runtime errors reported when using long input sequence lengths with LoRA

### PR DESCRIPTION
**Pull request by vivekgoe**
This PR has following fixes,

-  Increase size of indices tensors used to maintain multi-lora state information from max_num_batched_tokens to 3*max_num_batched_tokens. This increase is done to provide buffer for padding done in batch & sequence dimensions.

-  Move logic to remove padding from lora_logits from execute_model() back to Class LogitsProcessorWithLoRA, this is done to fix race condition caused by updating multi-lora state information directly.

FIX https://github.com/HabanaAI/vllm-fork/issues/237

### CODE REVIEW

**Review by michalkuligowski — APPROVED** [review]

File: vllm/worker/habana_model_runner.py [hunk 1]
```diff
@@ -1937,14 +1937,6 @@ def execute_model(
             )
```
Comment (michalkuligowski) [inline vllm/worker/habana_model_runner.py:1937]: In habana_main version you change also:
1189 + is_prefill=(num_prefills > 0):
1355 + is_prefill=is_prompt)
is this not needed here?
